apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/template/edit.js

```diff
@@ -23,8 +24,29 @@ import { addQueryArgs } from '@wordpress/url';
  */
 import './style.scss';
 
+// Hide the Block Sidebar when the Template block is selected
+domReady( () => {
```

Comment:
Can we see if we can add a better selector instead of targeting last-child with the addFilter hook. See example: https://github.com/Automattic/wp-calypso/blob/8e51dbeb7383bd3e40dec2904cc78bf3d30ba23d/apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/post-content/index.js#L32 

We can also query sidebar state, instead of debouncing this to apply another class. See https://github.com/WordPress/gutenberg/blob/15f5ff44609df48a47dee1d3e750470cdda50a5e/packages/edit-post/src/store/selectors.js#L25

Feel free to ping me if you need more info on ^^

Reply:
I haven't found any filters on the external editor wrapper that could be used for this, but only ones deeper in the DOM, closer to the blocks list (e.g. if I haven't misunderstood `editor.BlockListBlock`, it targets the actual block, even if the name could be misleading).
In this case, we would need, for example, to add a `.is-template-selected` class to, say, the body, the editor wrapper, or anything _outside_ the blocks list, in order to target the sidebar.

The `:last-child` in itself should be safe enough, considering that the sidebar header is an hardcoded list with only two elements (Document and Block), and we always need to hide the second one.

Though, maybe I'm either overthinking this or totally misunderstanding your suggestion. I'll double check if I've missed something obvious, or maybe do some lateral thinking, or eventually just ping you 😄

Reply:
It doesn't look like there is a good alternative for the last-child selector, but we can listen to the edit-post store and block-editor store:
<img width="498" alt="Screen Shot 2019-08-19 at 6 40 41 PM" src="https://user-images.githubusercontent.com/1270189/63310862-21962a00-c2b1-11e9-8b82-715928ad3fba.png">

<img width="1100" alt="Screen Shot 2019-08-19 at 6 43 05 PM" src="https://user-images.githubusercontent.com/1270189/63310908-4a1e2400-c2b1-11e9-9801-2b6e505eb5f7.png">

So sketching this out roughly in code this might look like:
```jsx
import { withSelect } from '@wordpress/data';

//wrap an appropriate component
withSelect( ( select ) => {
    const selectedBlock = select( 'core/block-editor' ).getSelectedBlock();
    const a8cTemplateSelected = selectedBlock && selectedBlock.name === 'a8c/template';
    const blockOptionsSelected = select( 'core/edit-post' ).getActiveGeneralSidebarName() === 'edit-post/block';
    return {
        'hideBlockSidebarTab': a8cTemplateSelected && blockOptionsSelected,
    }
} );
```

Two options then on how to use it, if we are able to add a modifier class in the right spot, we can add a classname + CSS
```jsx
//in an appropriate component render, let's add another classname:
render() {
    const { hideBlockSidebarTab } = this.props;
    return <container className={ hideBlockSidebarTab ? 'my_container.isTemplateSelected' : 'my_container' } />;
}

// then in CSS add a selector that targets this state to display:none;
```

Otherwise we can try and use https://reactjs.org/docs/hooks-effect.html to detect this transition state from false/true and trigger the CSS toggle on blockSidebarButton